tests/test_hash.py

```diff
@@ -122,14 +122,14 @@ def test_model_dump(bytes32str):
     model = Model.from_single(5)
     actual = model.model_dump()
     expected = {
-        "valuebytes8": "0x0000000000000005",
+        "valuebytes8": "0000000000000005",
         "valuestr8": "0x0000000000000005",
-        "valuebytes16": "0x00000000000000000000000000000005",
-        "valuebytes20": "0x0000000000000000000000000000000000000005",
+        "valuebytes16": "00000000000000000000000000000005",
+        "valuebytes20": "0000000000000000000000000000000000000005",
         "valuestr16": "0x00000000000000000000000000000005",
-        "valuebytes32": "0x0000000000000000000000000000000000000000000000000000000000000005",
```

Comment:
this is what i was hoping:

* even though HexBytes.hex() no longer has he 0x prefix, which I think is fine, we still can have the model have 0x when in str form like this via our custom validator.
* but hex() being different is the breaking change i want to accept, so maybe cannot rely on it so much anymore..